roomserver/storage/shared/storage.go

```diff
@@ -1445,6 +1446,54 @@ func (d *Database) ForgetRoom(ctx context.Context, userID, roomID string, forget
 	})
 }
 
+// PurgeRoom removes all information about a given room from the roomserver.
+// For large rooms this operation may take a considerable amount of time.
+func (d *Database) PurgeRoom(ctx context.Context, roomID string) error {
+	return d.Writer.Do(d.DB, nil, func(txn *sql.Tx) error {
+		roomNID, err := d.RoomsTable.SelectRoomNIDForUpdate(ctx, txn, roomID)
+		if err != nil {
+			if err == sql.ErrNoRows {
+				return fmt.Errorf("room %s does not exist", roomID)
+			}
+			return fmt.Errorf("failed to lock the room: %w", err)
+		}
+		if err = d.Purge.PurgeStateBlocks(ctx, txn, roomNID); err != nil {
+			return fmt.Errorf("failed to purge state blocks: %w", err)
+		}
+		if err = d.Purge.PurgeStateSnapshots(ctx, txn, roomNID); err != nil {
+			return fmt.Errorf("failed to purge state snapshots: %w", err)
+		}
+		if err = d.Purge.PurgeInvites(ctx, txn, roomNID); err != nil {
+			return fmt.Errorf("failed to purge invites: %w", err)
+		}
+		if err = d.Purge.PurgeMemberships(ctx, txn, roomNID); err != nil {
```

Comment:
I don't understand why this isn't just `d.Purge.PurgeRoom(ctx, txn, roomNID)`. We've added a bazillion functions here which end up being easy to forget one, and hard to verify they are all being called. Just combine all these SQL statements into a single function?